### PR TITLE
Fix 'Wconversion' warns: casting ints and check tcp endpoints

### PR DESCRIPTION
**Pull request by melvinhe**
*Issue number of the reported bug or feature request: #87 *

**Describe your changes**

Maintaining long value returned by 'strtol' and casts to int if port in reasonable range. The changed files diffs below show the changes in https://github.com/bloomberg/blazingmq/pull/214 possibly due to resetting my local wconv_fix to remote.

This particular PR deals with the following -Wconversion warnings that pollutes the build logs:

```
/home/runner/work/blazingmq/blazingmq/src/groups/mwc/mwcio/mwcio_channel.cpp:66:50: warning: conversion from ‘BloombergLP::bslstl::StringRefImp<char>::size_type’ {aka ‘long unsigned int’} to ‘int’ may change value [-Wconversion]
   66 |                                        str.length())) {                       \
      |                                        ~~~~~~~~~~^~
/home/runner/work/blazingmq/blazingmq/src/groups/mwc/mwcio/mwcio_channel.cpp:71:5: note: in expansion of macro ‘CHECKVALUE’
   71 |     CHECKVALUE(LOW_WATERMARK)
      |     ^~~~~~~~~~
/home/runner/work/blazingmq/blazingmq/src/groups/mwc/mwcio/mwcio_channel.cpp:66:50: warning: conversion from ‘BloombergLP::bslstl::StringRefImp<char>::size_type’ {aka ‘long unsigned int’} to ‘int’ may change value [-Wconversion]
   66 |                                        str.length())) {                       \
      |                                        ~~~~~~~~~~^~
/home/runner/work/blazingmq/blazingmq/src/groups/mwc/mwcio/mwcio_channel.cpp:72:5: note: in expansion of macro ‘CHECKVALUE’
   72 |     CHECKVALUE(HIGH_WATERMARK)
      |     ^~~~~~~~~~
```

```
/home/runner/work/blazingmq/blazingmq/src/groups/mwc/mwcio/mwcio_tcpendpoint.cpp:93:25: warning: conversion from ‘long int’ to ‘int’ may change value [-Wconversion]
   93 |     d_port = bsl::strtol(uri.c_str() + colon + 1, 0, 10);
      |              ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/blazingmq/blazingmq/src/groups/mwc/mwcio/mwcio_tcpendpoint.cpp: In member function ‘void BloombergLP::mwcio::TCPEndpoint::fromUriRaw(const string&)’:
/home/runner/work/blazingmq/blazingmq/src/groups/mwc/mwcio/mwcio_tcpendpoint.cpp:114:25: warning: conversion from ‘long int’ to ‘int’ may change value [-Wconversion]
  114 |     d_port = bsl::strtol(uri.c_str() + separator + 1, 0, 10);
```

**Testing performed**
N/A

**Additional context**
Currently only checks that 'temp_port' is within reasonable INT range rather than being more fine grained regarding valid tcp port values. This is likely fine though as we previously did not do a fine-grained valid port checking here.

### CODE REVIEW

**Comment by melvinhe** [comment]
@678098 Thanks for the detailed feedback; this really helps me write cleaner/scalable code and become a better (open source) engineer. 

I’ve updated your code suggestions in the most recent commit, which seems to be passing the test checks now. For future potential commits, I’ll just sync/fetch from main and create time stamped branches for each PR.

File: src/groups/mwc/mwcio/mwcio_tcpendpoint.cpp
Comment (678098) [inline src/groups/mwc/mwcio/mwcio_tcpendpoint.cpp:97]: According to `std::strtol` docs ( https://cplusplus.com/reference/cstdlib/strtol/ ) , `strtol` returns 0 either when conversion failed or when conversion succeeded with string `"0"`. These 2 cases are hard to distinguish so I propose to exclude 0 value for simplicity, so my initial comment was not good

```suggestion
    const long port = bsl::strtol(uri.c_str() + colon + 1, 0, 10);
    // For simplicity, do not accept ambiguous `port` value 0
    if (port <= 0 || port > 65535) {
        return false;  // RETURN
    }
    d_port = static_cast<int>(port);
```

`bsl::stoi` implementation uses `strtol`, and you can see the complexity of checks around it:
https://github.com/bloomberg/bde/blob/a30b9d4cb50ade5ea0b2ffded7cff60fa59c0112/groups/bsl/bslstl/bslstl_string.cpp#L193

File: src/groups/mwc/mwcio/mwcio_tcpendpoint.cpp
Comment (678098) [inline src/groups/mwc/mwcio/mwcio_tcpendpoint.cpp:118]: Here, let's do the same and add a check:

```suggestion
    if (separator == bsl::string::npos) {
        return;  // RETURN
    }

    const long port = bsl::strtol(uri.c_str() + separator + 1, 0, 10);
    // For simplicity, do not accept ambiguous `port` value 0
    if (port <= 0 || port > 65535) {
        return;  // RETURN
    }
    
    d_port = static_cast<int>(port);
```

**Review by 678098 — CHANGES_REQUESTED** [review]
Hi @melvinhe, sorry for the delay.
I prepared a couple of comments, I think we can apply them and merge the PR soon.

**Comment by melvinhe** [comment]
Hi @678098, changes for handling '0' port values have been applied. Passing all tests except build for macosx_arm64, which seem to be unrelated to the committed code changes.

**Review by 678098 — APPROVED** [review]

**Comment by 678098** [comment]
> Hi @678098, changes for handling '0' port values have been applied. Passing all tests except build for macosx_arm64, which seem to be unrelated to the committed code changes.

Thank you @melvinhe! Will merge this branch once the CI fix is merged

**Pull request merged**
